### PR TITLE
Raise an exception if the given parameter is not recognized and is longer than 2 characters

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

Address #256.

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/helpers/utils.py [hunk 1]
```diff
@@ -172,8 +172,16 @@ def build_arg_string(kwargs):
...
+    >>> print(build_arg_string(dict(R="1/2/3/4", J="X4i", watre=True)))
+    Traceback (most recent call last):
+      ...
+    pygmt.exceptions.GMTInvalidInput: Unrecognized parameter 'watre'.
     """
     gmt_args = []
+    # raise an exception for unrecognized options
+    for key in kwargs:
+        if len(key) > 2:
+            raise GMTInvalidInput(f"Unrecognized parameter '{key}'.")
```
Comment (weiji14) [inline pygmt/helpers/utils.py:184]: Why not put this in the `for key in filtered_kwargs` loop below?

Comment (seisman) [reply pygmt/helpers/utils.py:184]: It's OK to put it in the `for key in filtered_kwargs` loop, but it won't catch typos like `watre=False`.

File: pygmt/helpers/utils.py [hunk 2]
```diff
@@ -172,8 +172,16 @@ def build_arg_string(kwargs):
...
     # Exclude arguments that are None and False
```
Comment (seisman) [inline pygmt/helpers/utils.py:185]: Perhaps we can rewrite the whole function like below, so we don't need to loop over `kwargs` three times.
```python
    for key in filtered_kwargs:
        # raise an exception for unrecognized options
        if len(key) > 2:
            raise GMTInvalidInput(f"Unrecognized parameter '{key}'.")       
        if (kwargs[key] is None or kwargs[key] is False):
            pass  # Exclude arguments that are None and False
        elif is_nonstr_iter(kwargs[key]):
            for value in kwargs[key]:
                gmt_args.append(f"-{key}{value}")
        elif kwargs[key] is True:
            gmt_args.append(f"-{key}")
        else:
            gmt_args.append(f"-{key}{kwargs[key]}")
    return " ".join(sorted(gmt_args))
```

Comment (weiji14) [reply pygmt/helpers/utils.py:185]: Yep, looks good. It might conflict with #1487 but we can fix it later.

**Review by weiji14 — APPROVED** [review]
I had a last minute thought that the `len(key) > 2` check should probably go in the `@use_alias` decorator, but I really like this refactor so let's just go with this :stuck_out_tongue_closed_eyes:

**Review by willschlitzer — APPROVED** [review]

**Pull request merged**